### PR TITLE
Unfair Aura Immunity

### PR DESCRIPTION
**Pull request by dunjeon**
A feature that gives players (configurable) immunity auras against some spells in dungeons and raids.
Available Unfair List:  Charm, Fear, Root, Snare, Stun, Polymorph, Daze

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
...
 ## AutoBalanceModule
 - Latest build status with azerothcore: [![Build Status](https://travis-ci.org/azerothcore/mod-autobalance.svg?branch=master)](https://travis-ci.org/azerothcore/mod-autobalance)
-
-This module is intended to scale based on number of players, instance mobs and bosses' health, mana, and damage.
+* Script Name: AutoBalance
+* Original Authors: KalCorp and Vaughner
+* Original Script Name: AutoBalance
+* Description: This script is intended to scale based on number of players, instance mobs & world bosses' health, mana, and damage.
+* Ported to AzerothCore as a module
+* For installation instructions follow same as https://github.com/azerothcore/mod-duelreset
```
Comment (BarbzYHOOL) [inline README.md]: can you revert these changes to the README? Yehonal removed them on purpose
We will have to update the readme later anyway

File: acore-module.json [hunk 2]
```diff
@@ -1,9 +0,0 @@
-{
-  "name": "mod-autobalance",
-  "version": "2.0.0",
-  "compatibility" : [
-      { "version": "1.0.0", "branch": "none" },
-      { "version": "2.0.0", "branch": "2.0" },
-      { "version": "3.0.0-dev", "branch": "master" }
-  ]
-}
```
Comment (BarbzYHOOL) [inline acore-module.json]: @Yehonal the dependency system doesn't work yet so, should he delete this?

File: conf/AutoBalance.conf.dist [hunk 3]
```diff
@@ -222,9 +222,38 @@ AutoBalance.reward.dungeonToken = 47241
...
+#        Stun:        80869
+#        Polymorph:   80870
+#        Daze:        80871
+#
+#        Default:     "80865,80866,80867,80868,80869,80870,80871"
+
+AutoBalance.uai.enable = 1
+AutoBalance.uai.PlayerCount = 1
+AutoBalance.uai.AuraList = "80865,80866,80867,80868,80869,80870,80871" 
+#
```
Comment (BarbzYHOOL) [inline conf/AutoBalance.conf.dist]:
```suggestion
#

```

:D

**Comment by BarbzYHOOL** [comment]
ok we wil clean the useless files before mergin

btw what do you think about simply changing "uai" to a more semantically useful "UnfairAuraImmunity" or "UnfairAuras" ?

just a detail

I will test when travis is okay, notify me

**Comment by dunjeon** [comment]
@BarbzYHOOL  I defeated Travis!

File: sql/README.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+## Rules
+
+- Use `InnoDB` as engine for dynamic tables (most likely in the `auth` and `characters` databases).
+- Use `MyISAM` for **read-only** tables (in `world` database), but if you're not sure, just use innoDB.
+- Use `utf8` as charset.
+
+
+## Resources
+
+https://www.w3schools.com/sql/sql_datatypes.asp
```
Comment (BarbzYHOOL) [inline sql/README.md]: rofl you added it back xD

but NP we will sort it at the end

File: sql/world/uai_WorldSpellDBC.sql [hunk 5]
```diff
@@ -0,0 +1,8 @@
+use world;
```
Comment (BarbzYHOOL) [inline sql/world/uai_WorldSpellDBC.sql]:
```suggestion
```
Do not put the "use db" in SQL files, most people will have a different DB name than world :)

File: sql/world/uai_WorldSpellDBC.sql [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+INSERT INTO `spell_dbc` VALUES (80868,0,0,256,268468224,0,0,0,0,0,0,0,0,0,1,0,0,101,0,0,0,0,21,1,0,-1,0,0,6,0,0,1,0,0,0,0,0,0,0,0,0,0,0,25,0,0,0,0,0,0,0,0,77,0,0,0,0,0,0,0,0,0,0,0,11,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,1,0,1,'AutoBalance Immune Snare');
+INSERT INTO `spell_dbc` VALUES (80869,0,0,256,268468224,0,0,0,0,0,0,0,0,0,1,0,0,101,0,0,0,0,21,1,0,-1,0,0,6,0,0,1,0,0,0,0,0,0,0,0,0,0,0,25,0,0,0,0,0,0,0,0,77,0,0,0,0,0,0,0,0,0,0,0,12,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,1,0,1,'AutoBalance Immune Stun');
+INSERT INTO `spell_dbc` VALUES (80870,0,0,256,268468224,0,0,0,0,0,0,0,0,0,1,0,0,101,0,0,0,0,21,1,0,-1,0,0,6,0,0,1,0,0,0,0,0,0,0,0,0,0,0,25,0,0,0,0,0,0,0,0,77,0,0,0,0,0,0,0,0,0,0,0,17,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,1,0,1,'AutoBalance Immune Polymorph');
+INSERT INTO `spell_dbc` VALUES (80871,0,0,256,268468224,0,0,0,0,0,0,0,0,0,1,0,0,101,0,0,0,0,21,1,0,-1,0,0,6,0,0,1,0,0,0,0,0,0,0,0,0,0,0,25,0,0,0,0,0,0,0,0,77,0,0,0,0,0,0,0,0,0,0,0,27,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,1,0,1,'AutoBalance Immune Daze');
```
Comment (BarbzYHOOL) [inline sql/world/uai_WorldSpellDBC.sql]:
```suggestion
INSERT INTO `spell_dbc` VALUES (80871,0,0,256,268468224,0,0,0,0,0,0,0,0,0,1,0,0,101,0,0,0,0,21,1,0,-1,0,0,6,0,0,1,0,0,0,0,0,0,0,0,0,0,0,25,0,0,0,0,0,0,0,0,77,0,0,0,0,0,0,0,0,0,0,0,27,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,1,0,1,'AutoBalance Immune Daze');

```
needed

File: sql/world/uai_WorldSpellDBC.sql (hunk 5 above)
Comment (dunjeon) [reply sql/world/uai_WorldSpellDBC.sql]: I would like to comment it out with a -- use world;  Ok?

File: sql/README.md (hunk 4 above)
Comment (dunjeon) [reply sql/README.md]: Yeah, that file is created by create_module.sh

Comment (dunjeon) [reply sql/README.md]: I used it to get the correct sql directory structure... I will get rid of it.

**Pull request closed without merging**